api/v1alpha2/resource_error.go

```diff
@@ -19,70 +19,150 @@
 
 package v1alpha2
 
+import (
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+)
+
+type ResourceErrorSeverity string
+type ResourceErrorType string
+
+const (
+	SeverityMinor ResourceErrorSeverity = "Minor"
```

Comment:
We should have comments here that define what these mean for people writing controllers, and what we expect Flux or Slurm to do with them.